packages/component/index.js

```diff
@@ -69,7 +69,7 @@ function bindUi(ui, el) {
 function bindComponents(components, el) {
   return Object.keys(components).reduce(
     (hash, key) => {
-      const {type, selector, options, ghost = false, optional = false, asArray = false} = components[key]
```

Comment:
I would not forward `options`.

EG: Component `A` has a child `B`, they both accept option called `speed`, whose default value is `500`.

You do:
```javascript
const components = {
  a: {
    type: A, 
    selector: '.a', 
    options: {speed: 100}, 
    components: {
      b: {
        type: B,
        selector: '.b'
      }
    }
  }
```
You might expect `B` to have a `speed` of `500` because you never declared a `speed` value and you want to use the default.
Instead `speed` is `100` in both components.
You never explicitly declared so, it's a side effect and it's not clear.